### DISCUSSION TITLE
LEmonLdap LLNG for login

### DISCUSSION THREAD

**Discussion by compgeniuses** [Lissy93/dashy#573] [Help and Support]
I can see that the application by defaults support Keycloack for authentication.
Does this application, or can it support [lemonldap ](https://lemonldap-ng.org/)for authentication instead of Keycloak, if i already have an instance of LLNG running?

Since the Two applications do essentially the same thing, how can the keycloack settings be adopted to suite well for LLNG?

Regards

**Comment by lissy93** [comment]
Not familiar with LemonLDAP, but if their API is truly a 1-for-1 match with KC, then it would work the same way. If not, you'll have to follow their docs to get it set up.

**Comment by compgeniuses** [comment]
What do you mean by API, does it make use of Direct API. what about standard SAML authentication parameters?

**Reply by lissy93** [reply]
SAML 2.0 is supported, but only [through KC](https://www.keycloak.org/docs/latest/server_admin/index.html#saml-v2-0-identity-providers). 

There's so many different auth providers and protocols, that it's not feasible for me to add native support for all of them, and LemonLDAP is not widely used. 
And afaik, SAML doesn't cover authorization, so would need to be done though KC (or another broker) for permissions anyway.  

You can of course put Dashy behind your own auth provider with a reverse proxy and not use the built-in auth. Or like i said before, fork the project and implement whatever you like. 

For more info, see the [Authentication Docs](https://github.com/Lissy93/dashy/blob/master/docs/authentication.md).

**Reply by lissy93** [reply]
> What do you mean by API 

By API, i mean [application programming interface](https://en.m.wikipedia.org/wiki/API), its the thing you use to interact with a service.

For example, here's how auth is done with KC api: https://github.com/Lissy93/dashy/blob/master/src/utils/KeycloakAuth.js
